main.js

```diff
@@ -1167,7 +1205,9 @@ function main() {
 
         // Just remember, that such a nodeID created
         nodes[nodeID] = {id: calcName(nodeID), ready: false, native: null};
-        extendNode(nodeID, {});
+        if (inclusion) {
+            extendNode(nodeID, {});
```

Comment:
Are you sure about this? Doesn't that mean the objects are not re-created if they were deleted within ioBroker?

Reply:
The node added event is called during inclusion (which is still covered), during startup (where it is superfluous as it just overwrites existing data just to revert it again in "node available") and during refresh where it is still covered by "node available".

Reply:
Ok so we still create the node objects if they are missing in the `node available` event?

Reply:
Yes that is correct